source/opt/dead_variable_elimination.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dead_variable_elimination.h"
+
+#include "reflect.h"
+
+namespace spvtools {
+namespace opt {
+
+Pass::Status DeadVariableElimination::Process(spvtools::ir::Module* module) {
+  bool modified = false;
+  module_ = module;
+  def_use_mgr_.reset(new analysis::DefUseManager(consumer(), module));
+  FindNamedOrDecoratedIds();
+  decoration_manager_ = std::unique_ptr<analysis::DecorationManager>(
+      new analysis::DecorationManager(module));
+
+  std::vector<uint32_t> ids_to_remove;
+
+  // Get the reference count for all of the global OpVariable instructions.
+  for (auto& inst : module->types_values()) {
+    if (inst.opcode() == SpvOp::SpvOpVariable) {
```

Comment:
I slightly prefer the form that leads to less indentation:

```
if (inst.opcode() != SpvOp::SpvOpVariable) {
   continue;
}
```

Reply:
done